src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/TdsParser.cs

```diff
@@ -2911,11 +2911,11 @@ private bool TryProcessDone(SqlCommand cmd, SqlDataReader reader, ref RunBehavio
             ushort status;
             int count;
 
-            // This is added back since removing it from here introduces regressions in Managed SNI.
-            // It forces SqlDataReader.ReadAsync() method to run synchronously,
-            // and will block the calling thread until data is fed from SQL Server.
-            // TODO Investigate better solution to support non-blocking ReadAsync().
-            stateObj._syncOverAsync = true;
+            if (LocalAppContextSwitches.MakeReadAsyncBlocking)
```

Comment:
Could you update description to enlist new AppContext switches applying to .NET Core with their default values too? We'll need that for documentation in release notes.

Reply:
Updated.